pkg/converter/tool/nydusd.go

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2022. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package tool
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type NydusdConfig struct {
+	EnablePrefetch bool
+	NydusdPath     string
+	BootstrapPath  string
+	ConfigPath     string
+	BackendType    string
+	BackendConfig  string
+	BlobCacheDir   string
+	APISockPath    string
+	MountPath      string
+}
+
+// Nydusd runs nydusd binary.
+type Nydusd struct {
+	NydusdConfig
+}
+
+type daemonInfo struct {
+	State string `json:"state"`
+}
+
+var configTpl = `
+{
+	"device": {
+		"backend": {
+			"type": "{{.BackendType}}",
+			"config": {{.BackendConfig}}
+		},
+		"cache": {
+			"type": "blobcache",
+			"config": {
+				"work_dir": "{{.BlobCacheDir}}"
+			}
+		}
+	},
+	"mode": "cached",
+	"iostats_files": false,
+	"fs_prefetch": {
+		"enable": {{.EnablePrefetch}},
+		"threads_count": 10,
+		"merging_size": 131072
+	},
+	"digest_validate": true,
```

Comment:
If it is to be used during build process, I'd suggest that we disable digest validation since it affects performance a lot. For tests, we can just enable it for sure. Or are you planning to add an option to set it later on?